File: typescript/15-ends-with-suffix.ts
// ¿Termina con sufijo? — Fácil · Strings
// Refactoriza el código para hacerlo más conciso y legible.
// Restricción: no uses .endsWith().
// Fuente: https://coding-challenges.dev/problems/typescript-termina-con-sufijo

export function endsWithSuffix(str: string, suffix: string): boolean {
  return suffix.length === 0 ||
    (suffix.length <= str.length && str.slice(-suffix.length) === suffix);
}

File: python/15_ends_with_suffix.py
"""¿Termina con sufijo? — Fácil · Strings

Determina si s termina con suffix. No uses str.endswith(); un sufijo vacío
siempre coincide y uno más largo que s nunca coincide.

Fuente: https://coding-challenges.dev/problems/python-termina-con-sufijo
"""


def ends_with_suffix(s: str, suffix: str) -> bool:
    return len(suffix) <= len(s) and s[len(s) - len(suffix):] == suffix
